libvast/src/system/catalog.cpp

```diff
@@ -245,7 +245,8 @@ catalog_state::lookup_impl(const expression& expr, const type& schema) const {
       auto extract_expr = detail::overload{
         [&](const meta_extractor& lhs,
             const data& d) -> catalog_lookup_result::candidate_info {
-          if (lhs.kind == meta_extractor::type) {
+          // TODO: What to do for `#schema_id` here?
```

Comment:
Good point regarding the tests. Added some with fbc491019973a387c03c4f43806594cf76be536f.

Regarding the implementation of `lookup_impl`: Please take a look at e3e02289fc16c72ca600750cc5123be5e363c3a8. I'd assume quite a low probability that this is what we actually want, but it's hard to tell.

Reply:
I think this is correct.

Reply:
Okay, then I think the implementation for `#schema` could be made a lot more clear. I was quite confused by what's going on, but IIRC, this is because the partition story got simplified, so this is probably just legacy code. Anyway, thanks for taking a look!

Reply:
Yeah, this function used to work on more than one schema.